DependencyInjection/Configuration.php

```diff
@@ -69,6 +69,26 @@ public function getConfigTreeBuilder()
                     return $v;
                 })
             ->end()
+            ->beforeNormalization() // @deprecated toolbar in 1.3.0, use profiling instead
+                ->ifTrue(function ($v) {
+                    return is_array($v) && array_key_exists('toolbar', $v) && is_array($v['toolbar']);
+                })
+                ->then(function ($v) {
+                    if (array_key_exists('enabled', $v['toolbar']) && 'auto' === $v['toolbar']['enabled']) {
+                        $v['toolbar']['enabled'] = $this->debug; // @deprecated value auto in 1.3.0
+                    }
+
+                    if (array_key_exists('profiling', $v) && is_array($v['profiling'])) {
```

Comment:
imo the BC promise covers that you can update the bundle and it continues to work as before. but once you add new configuration that was not possible before the update, it is ok to complain when you leave the old config in place. merging old and new config feels strange, it would mean you can configure both at the same time.

with auto config i meant `->defaultValue()` places. i think those come after beforeNormalization - if i am wrong and they are already present, we indeed would need to merge.